main.cc

```diff
@@ -545,6 +545,84 @@ static locator::host_id initialize_local_info(sharded<db::system_keyspace>& sys_
     return linfo.host_id;
 }
 
+static bool raft_topology_change_enabled(const db::config& cfg) {
+    return cfg.consistent_cluster_management() &&
+        cfg.check_experimental(db::experimental_features_t::feature::CONSISTENT_TOPOLOGY_CHANGES);
+}
+
+static void initialize_system_keyspace(sharded<db::system_keyspace>& sys_ks,
+        sharded<cql3::query_processor>& qp,
+        sharded<locator::snitch_ptr>& snitch,
+        distributed<replica::database>& db,
+        sharded<locator::effective_replication_map_factory>& erm_factory,
+        const db::config& cfg,
+        sharded<gms::feature_service>& feature_service)
+{
+    supervisor::notify("starting system keyspace");
+    sys_ks.start(std::ref(qp), std::ref(db), std::ref(snitch)).get();
```

Comment:
No, no, any `sharded<service>.start(args...)` call should remain in main(). The ultimate goal (that's almost reached in cql_test_env's do_with()) is to make main look like

```
    sharded<service> srv;
    co_await srv.start(deps...);
    auto stop_srv = defer([&srv] { co_await srv.stop(); });
    co_await srv.invoke_on_all(&service::start);

    ... // same sequence for other services
```

all the other stuff, like dist_loader::init_sys_ks(), maybe_init_schema_commitlog(), etc you can scatter the weay you like

Reply:
why?

Reply:
presumably because of RAII for stopping.

Reply:
Yes, because of `auto stop = defer([&srv] { co_await srv.stop(); });` thing. Thank you, Tomek